.github/workflows/pull_request.yml

```diff
@@ -26,14 +26,31 @@ jobs:
       linux_nightly_6_0_arguments_override: "--explicit-target-dependency-import-check error -Xswiftc -require-explicit-sendable"
       linux_nightly_main_arguments_override: "--explicit-target-dependency-import-check error -Xswiftc -require-explicit-sendable"
 
-  examples:
+  construct-examples-matrix:
+    name: Construct Examples matrix
+    runs-on: ubuntu-latest
+    outputs:
+      examples-matrix: '${{ steps.generate-matrix.outputs.examples-matrix }}'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - id: generate-matrix
+        run: echo "examples-matrix=$(curl -s https://raw.githubusercontent.com/apple/swift-nio/main/scripts/generate_matrix.sh | bash)" >> "$GITHUB_OUTPUT"
+        env:
+          MATRIX_LINUX_5_9_ENABLED: false
+          MATRIX_LINUX_5_10_ENABLED: false
+          MATRIX_LINUX_COMMAND: "./dev/build-examples.sh"
+          MATRIX_LINUX_SETUP_COMMAND: ""
```

Comment:
I _think_ you can omit this if you want

Reply:
The examples are about to move to the SwiftPM build plugin so we'll need to install `protoc` as a setup. I left it blank so I didn't have to find the name of the env variable again 🙃